python/lsst/ts/mtaos/model.py

```diff
@@ -1059,46 +1063,43 @@ async def _poll_butler_outputs(
         """
         self.log.debug("Polling butler for WEP outputs.")
 
-        butler = dafButler.Butler(self.data_path)
+        butler = dafButler.Butler(self.data_path, collections=[run_name])
         start_time = time.time()
+        elapsed_time = 0.0
 
-        while True:
-            elapsed_time = time.time() - start_time
-
+        while elapsed_time < timeout:
             try:
-                datasetRefs = list(
-                    butler.registry.queryDatasets(
-                        datasetType="postISRCCD", collections=[run_name]
-                    )
+                self.log.info(
+                    f"Querying datasets: zernike_table_name={self.zernike_table_name}, "
+                    f"{run_name=} {extra_id=}."
```

Comment:
Personally I don't love and info level log inside the polling loop, but maybe that's OK/a feature in T&S code?